src/Components/AuthShared/ChangePasswordModel/ChangePasswordModel.tsx

```diff
@@ -0,0 +1,110 @@
+import { Button, Modal } from "flowbite-react";
+import { useEffect, useState } from "react";
+import logo from "../../../assets/logo.png";
+import PasswordInput from "../PasswordInput/PasswordInput";
+import { SubmitHandler, useForm } from "react-hook-form";
+import ButtonForm from "../ButtonForm/ButtonForm";
+import {
+  ChangePasswordReguest,
+  ChangePasswordResponse,
+} from "../../../InterFaces/Interfaces";
+import toast from "react-hot-toast";
+import { apiClient, AUTHENTICATION_URLS } from "../../../Apis/EndPoints";
+import { AxiosError } from "axios";
+import { PasswordValidation } from "../../../Validation/Validation";
+
+export default function ChangePasswordModal() {
+  const [openModal, setOpenModal] = useState(false);
+  const {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    watch,
+    setFocus,
+  } = useForm();
+
+  const onSubmit: SubmitHandler<ChangePasswordReguest> = async (data) => {
+    const toastId = toast.loading("Processing...");
+    try {
+      const { Confirm_Password, ...submitData } = data;
+
+      const response = await apiClient.post<ChangePasswordResponse>(
+        AUTHENTICATION_URLS.changePassword,
+        submitData
+      );
+
+      setOpenModal(false);
+      toast.success("Paswword is changed Successfully", {
+        id: toastId,
+      });
+    } catch (error) {
+      const axiosError = error as AxiosError<{ message: string }>;
```

Comment:
we can handle the errors with their toasts in the interceptor as well.
We can make also a general function that checks the errors incrementally based on the backend responses that we've seen.